src/yunohost/log.py

```diff
@@ -0,0 +1,436 @@
+        self.file_handler = FileHandler(filename)
+        self.file_handler.formatter = Formatter('%(asctime)s: %(levelname)s - %(message)s')
+
+        # Listen to the root logger
+        self.logger = getLogger('yunohost')
+        self.logger.addHandler(self.file_handler)
+
+    def flush(self):
+        """
+        Write or rewrite the metadata file with all metadata known
+        """
+
+        filename = os.path.join(self.path, self.name + METADATA_FILE_EXT)
+        with open(filename, 'w') as outfile:
+            yaml.safe_dump(self.metadata, outfile, default_flow_style=False)
+
+    @property
+    def name(self):
+        """
+        Name of the operation
+        This name is used as filename, so don't use space
+        """
+        if self._name is not None:
+            return self._name
+
+        name = [self.started_at.strftime("%Y%m%d-%H%M%S")]
+        name += [self.operation]
+        if self.related_to:
+            if isinstance(self.related_to[0], tuple):
+                name += [self.related_to[0][1]]
+            else:
+                name += self.related_to[0].values()
+        self._name = '-'.join(name)
+        return self._name
+
+    @property
+    def metadata(self):
+        """
+        Dictionnary of all metadata collected
+        """
+
+        data = {
+            'started_at': self.started_at,
+            'operation': self.operation,
+        }
+        if self.related_to is not None:
+            data['related_to'] = self.related_to
+        if self.ended_at is not None:
+            data['ended_at'] = self.ended_at
+            data['success'] = self._success
+            if self.error is not None:
+                data['error'] = self._error
+                # TODO: detect if 'extra' erase some key of 'data'
+        data.update(self.extra)
+        return data
+
+    def success(self):
+        """
+        Declare the success end of the unit operation
+        """
+        self.close()
+
+    def error(self, error):
+        """
+        Declare the failure of the unit operation
+        """
+        return self.close(error)
+
+    def close(self, error=None):
+        """
+        Close properly the unit operation
+        """
+        if self.ended_at is not None or self.started_at is None:
+            return
+        if error is not None and not isinstance(error, basestring):
+            error = str(error)
+        self.ended_at = datetime.now()
+        self._error = error
+        self._success = error is None
+        if self.logger is not None:
+            self.logger.removeHandler(self.file_handler)
+
+        is_api = msettings.get('interface') == 'api'
+        desc = _get_description_from_name(self.name)
+        if error is None:
+            if is_api:
+                msg = m18n.n('log_link_to_log', name=self.name, desc=desc)
+            else:
+                msg = m18n.n('log_help_to_get_log', name=self.name, desc=desc)
+            logger.info(msg)
```

Comment:
I don't know what to do about this ... but printing this message after each command looks a bit too much... In particular, this also apply to "sub" operations. To illustrate, when adding a domain I get this : 

```
 > yunohost domain add test.lol
Administration password: 
Success! Successfully installed a self-signed certificate for domain test.lol!
Info: To view the log of the operation 'Install self signed certificate on 'test.lol' domain', use the command 'yunohost log display 20180725-144845-selfsigned_cert_install-test.lol'
Warning: The configuration file '/etc/nginx/conf.d/yunohost_admin.conf' has been manually modified and will not be updated
Success! The configuration has been updated for service 'dnsmasq'
Success! The configuration has been updated for service 'metronome'
Info: To view the log of the operation 'Regenerate system configurations 'nginx'', use the command 'yunohost log display 20180725-144847-service_regen_conf-nginx'
Success! The SSOwat configuration has been generated
Success! The domain has been created
Info: To view the log of the operation 'Add 'test.lol' domain into system configuration', use the command 'yunohost log display 20180725-144845-domain_add-test.lol'
```

So here I got three `To view the log of the operation [...]` about logs which I won't go checking because everything worked anyway ... As a user, what I really want to know is if everything worked correctly and if I need to do something in particular, so in that perspective those messages (at least the first two) looks like noise :/